src/email-only-signup.js

```diff
@@ -1,132 +1,284 @@
-};
-
-export function getResponseMsg(response, pageLocation) {
-	// Keep marketing copy somewhere
+/*
+ * convert a response 'code' to html message
+ * @param {String} response
+ * @param {String} pageLocation
+ * @returns {HTMLString}
+ */
+export function getResponseMsg (response, pageLocation) {
 	const responseMsg = {
 		'SUBSCRIPTION_SUCCESSFUL': 'Thanks – look out for your first briefing soon.',
 		'INVALID_REQUEST': 'Sorry, something went wrong. Please try again.',
 		'ALREADY_SUBSCRIBED': 'It looks like you’re currently receiving the daily top stories summary email. If you’re interested in getting access to more FT content, why not <a target="_blank" style="text-decoration:none;color:#27757B;" href="/products?segID=0801043" data-trackable="trial-existing">sign up to a 4 week Trial</a>?',
 		'USER_ARCHIVED': 'It looks like you’ve signed up to the daily top stories summary email before. If you’re interested in getting access to more FT content, why not <a target="_blank" style="text-decoration:none;color:#27757B;" href="/products?segID=0801043" data-trackable="trial-archived">sign up to a 4 week Trial</a>?',
 		'USER_NOT_ANONYMOUS': `It looks like you already have an account with us. <a href="/login?location=${pageLocation}" style="text-decoration:none;color:#27757B;" data-trackable="sign-in">Sign in</a>.`
 	};
-
 	return responseMsg[response] ? responseMsg[response] : responseMsg.INVALID_REQUEST;
 }
 
-export default {
-	init(el, options = {}) {
-		defaultsDeep(options, optionsFromData(el), defaultOptions);
-
-		const closeButton = el.querySelector('[data-o-email-only-signup-close]');
-		const lightSignupForm = el.querySelector('[data-o-email-only-signup-form]');
-		const displaySection = el.querySelector('[data-o-email-only-signup-completion-message]');
-		const emailField = el.querySelector('input[name=email]');
-		const topicSelect = el.querySelector('[data-o-email-only-signup-dropdown]');
-		const invalidEmailMessage = el.querySelector('[data-o-email-only-signup-email-error]');
-		const SELECT_INACTIVE_CLASS = 'o-email-only-signup__select--inactive';
-
-		const pageLocation = window.location.href;
-
-		// Handle user interaction
-		lightSignupForm.addEventListener('submit', (e) => {
-			e.preventDefault();
-
-			if (isValidEmail(emailField.value)) {
-				const opts = {
-					method: 'POST',
-					headers: {
-						'Content-type': 'application/x-www-form-urlencoded'
-					},
-					credentials: 'include',
-					body: serializeFormInputs(e.target)
-				};
-
-				fetch(options.signupUrl, opts)
-					.then(response => response.text())
-					.then(response => {
-				displaySection.innerHTML = getResponseMsg(response, pageLocation);
-					})
-					.catch(err => console.log(err));
-
-			} else {
-				toggleValidationErrors();
-			}
-		});
+/*
+ * Init component
+ * @param {Element} element - the element containing the form
+ * @options {Object} options - custom options
+ */
+export function init (element, options={}) {
```

Comment:
i'm surprised this builds, in general `export default { init() {}}` is not the same as `export function init`